src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,158 @@
+  const [isChecked, setIsChecked] = React.useState(
+    React.isValidElement(context?.checkbox)
+      ? context?.checkbox.props['checked']
+      : false,
+  );
+  const isSelected = React.useMemo(() => {
+    return context?.selectedNode === label;
+  }, [context?.selectedNode, label]);
+
+  return (
+    <li role='treeitem' aria-expanded={expanded} {...rest}>
+      <div
+        className={cx('iui-tree-node', {
+          'iui-active': isSelected,
+          'iui-disabled': isDisabled,
+          className,
+        })}
+        style={styleLevel}
+        onClick={(e) => {
+          if (e.currentTarget != e.target) {
+            return;
+          }
+          if (isSelected) {
+            context?.setSelectedNode?.('');
+          } else if (!isDisabled) {
+            context?.setSelectedNode?.(label);
+            context?.onNodeSelected?.();
+          }
+        }}
+      >
+        {context?.checkbox && React.isValidElement(context?.checkbox)
+          ? React.cloneElement(context?.checkbox, {
+              className: 'iui-tree-node-checkbox',
+              disabled: isDisabled,
+              checked: isChecked,
+              onClick: () => {
+                setIsChecked(!isChecked);
+                context.onNodeCheckboxSelected?.();
+              },
+            })
+          : context?.checkbox}
+        <div className='iui-tree-node-content'>
+          {children && (
+            <IconButton
+              styleType='borderless'
+              size='small'
+              onClick={() => {
+                setExpanded(!expanded);
+                context?.onNodeExpanded?.();
+              }}
+              disabled={isDisabled}
+            >
+              <SvgChevronRight
+                className={cx('iui-tree-node-content-expander-icon', {
+                  'iui-tree-node-content-expander-icon-expanded': expanded,
+                })}
+              />
+            </IconButton>
+          )}
+          {icon &&
+            React.cloneElement(icon, {
+              className: cx('iui-tree-node-content-icon', icon.props.className),
+            })}
+          <span className='iui-tree-node-content-label'>
+            <div className='iui-tree-node-content-title'>{label}</div>
+            <div className='iui-tree-node-content-caption'>{sublabel}</div>
+          </span>
+        </div>
+      </div>
+      {children && expanded && (
+        <ul className='iui-sub-tree' role='group'>
+          <TreeContext.Provider
+            value={{
+              selectedNode: context?.selectedNode,
+              setSelectedNode: context?.setSelectedNode,
+              onNodeSelected: context?.onNodeSelected,
+              onNodeExpanded: context?.onNodeExpanded,
+              checkbox: context?.checkbox,
+              onNodeCheckboxSelected: context?.onNodeCheckboxSelected,
+              nodeDepth: currentDepth + 1,
+            }}
+          >
+            {React.Children.map(children, (node) => node)}
```

Comment:
You no longer need to map as you don't provide any props or anything.

Reply:
Removed